src/ensembl/src/shared/components/instant-download/instant-download-fetch/fetchForProtein.ts

```diff
@@ -20,42 +20,61 @@ import {
   ProteinOption,
   proteinOptionsOrder
 } from 'src/shared/components/instant-download/instant-download-protein/InstantDownloadProtein';
+import {
+  fetchSequenceChecksums,
+  ProductGeneratingContextFragment
+} from './fetchSequenceChecksums';
 
 type FetchPayload = {
+  genomeId: string;
   transcriptId: string;
   options: ProteinOptions;
 };
 
 export const fetchForProtein = async (payload: FetchPayload) => {
-  const { transcriptId, options } = payload;
-  const urls = buildUrlsForProtein(transcriptId, options);
+  const { genomeId, transcriptId, options } = payload;
+  const productGeneratingContext = await fetchSequenceChecksums({
+    genomeId,
+    transcriptId
+  });
 
+  const urls = buildUrlsForProtein(productGeneratingContext, options);
   const sequencePromises = urls.map((url) =>
     fetch(url).then((response) => response.text())
   );
 
   const sequences = await Promise.all(sequencePromises);
-  const combinedFasta = sequences.join('\n');
+  const combinedFasta = sequences.join('\n\n');
 
   downloadAsFile(combinedFasta, `${transcriptId}.fasta`, {
     type: 'text/x-fasta'
   });
 };
 
-const buildUrlsForProtein = (id: string, options: ProteinOptions) => {
+const buildUrlsForProtein = (
+  productGeneratingContext: ProductGeneratingContextFragment,
+  options: ProteinOptions
+) => {
   return options
     ? proteinOptionsOrder
         .filter((option) => options[option])
-        .map((option) => buildFetchUrl(id, option))
+        .map((option) => buildFetchUrl(productGeneratingContext, option))
     : [];
 };
 
-const buildFetchUrl = (id: string, sequenceType: ProteinOption) => {
-  const sequenceTypeToTypeParam: Record<ProteinOption, string> = {
-    proteinSequence: 'protein',
+const buildFetchUrl = (
+  productGeneratingContext: ProductGeneratingContextFragment,
+  sequenceType: ProteinOption
+) => {
+  const sequenceTypeToContextType: Record<ProteinOption, string> = {
+    proteinSequence: 'product',
     cds: 'cds'
   };
-  const typeParam = sequenceTypeToTypeParam[sequenceType];
+  const contextType = sequenceTypeToContextType[
+    sequenceType
+  ] as keyof ProductGeneratingContextFragment;
+  const sequenceChecksum =
+    productGeneratingContext[contextType]?.sequence_checksum;
 
-  return `https://rest.ensembl.org/sequence/id/${id}?content-type=text/x-fasta&type=${typeParam}`;
+  return `http://refget.review.ensembl.org/refget/sequence/${sequenceChecksum}?accept=text/plain`;
```

Comment:
1) For the time being, we should request the sequence in the fasta format (`accept=text/x-fasta` rather than accept=text/plain). This won't work when we start getting genomic sequences from ENA; but by that time, we'll think of something.
2) Remember to update the host to point to an actual refget server available from outside Ensembl rather than to a feature deployment of refget proxy.

Reply:
#1 is done. Will look into the other two when the PR is approved.

Reply:
Points 2 and 3 have transformed into just using `/refget`, no hostname necessary.

Webpack dev server config will have to be updated here: https://github.com/Ensembl/ensembl-client/blob/dev/src/ensembl/webpack/environments/webpack.config.dev.js#L48 to include the `/refget` endpoint and point to the staging server.

(plus a reminder to change the `accept` query parameter in the other fetcher file to text/x-fasta)

Reply:
✅